### ISSUE TITLE
Create better SQL resources for names.csv

### ISSUE THREAD

**Issue by mdahlman** [carltonnorthern/nicknames#42]
I'll start by saying that having `names-mysql.sql` is far better than not having it. Thanks to the guys that created it.
But there are a few aspects I don't like about it. I'm thinking of adding some improvements. **I would very much like feedback from others about what would be most useful.** Here's my mini-spec:

- A SQL neutral (ANSI) version of the file would be appropriate here rather than MySQL or other technology-specific version.
- Include a version exactly parallel to names.csv. i.e. it would have exactly the same number of rows.
- Include a normalized version. Following the model of `names-mysql.sql` available today.
- Include a python script for regenerating these files whenever `names.csv` is updated.

Ideas for the file names:
- `names.sql`
- `names-normalized.sql`
- `generate-sql.py`

**Comment by NickCrews** [comment]
I admit I do not use SQL much and so don't have strong opinions here. being SQL dialect agnostic sounds great, but I can't speak to if there is some drawback. This case seems so simple it seems this shouldn't be that hard of a goal.

I think writing a script to auto-generate from the csv file is key. 

I'm not sure what you mean by "a version exactly parrallel to names.csv". The table in SQL needs a constant number of columns, right? So how do you deal with the variable-number-of-nicknames-per-line in the CSV?

I see the only end use cases being analogous to the python version, with `nicknames_of(name)` and `canonicals_of(name)`. The normalized version does this. What benefit would the un-normalized version give?

python scrips should have underscores, so should be `generate_sql.py`. Besides that names seem fine. IDK sql naming standards, but maybe `populate_nickname_table.sql` could be more accurate, to say what its doing as a verb?

**Comment by mdahlman** [comment]
The normalized version is, probably, the one most folks will use, as you suggest.

The use case for the version in the same format as `names.csv` is to make it very easy to say, "What is this csv resource? I'd like to explore it." Some folks might like to paste it into a spreadsheet. Some folks [like me] would like to explore it in a relational database using SQL.

Good point: I'll use underscores.

Actually, it would be more accurate to call the file `create_and_populate_nickname_normalized_table.sql`. But that's quite a mouthful. So I'm inclined to stay with `nickname_normalized.sql`. (Until someday if there would be a reason for more precision.)

**Comment by NickCrews** [comment]
@mdahlman did a great job in #48, I just need to follow up and add a CI check, and then we can close this!

**Comment by NickCrews** [comment]
Thank you @mdahlman !

**Comment by mdahlman** [comment]
I was pleased to be able to offer the SQL structure.
I really appreciate you turning it into _real_ python code. I will review it carefully.
Could you elaborate on the comment, "finding it hard to stay DRY"?

**Comment by NickCrews** [comment]
eh, I don't don't know about "real" :) I thought yours was much simpler, adding the output-path-but-also-with-default feature really made it ugly.

DRY as in Don't Repeat Yourself. I needed to be able to call two separate runs from the command line eg `generate.py --type normalized` as well as `generate.py --type nicknames`, which meant that I needed to split apart the main() into two separate methods. this was leading to a lot of duplicated code using the otherwise-I-like-it-more functional style, so I had to adopt the more-complicated-but-less-duplicated polymorphic implementation.
